app-commons/src/main/scala/org/bitcoins/commons/jsonmodels/dlc/CETCalculator.scala

```diff
@@ -0,0 +1,398 @@
+package org.bitcoins.commons.jsonmodels.dlc
+
+import org.bitcoins.core.currency.Satoshis
+import org.bitcoins.core.protocol.tlv.{
+  DLCOutcomeType,
+  EnumOutcome,
+  UnsignedNumericOutcome
+}
+import org.bitcoins.core.util.NumberUtil
+
+import scala.annotation.tailrec
+
+object CETCalculator {
+
+  /** Given a range and a payout function with which to build CETs,
+    * the first step is to split the range into sub-ranges which
+    * can be compressed or cannot be compressed, represented here
+    * as CETRanges.
+    *
+    * These ranges are inclusive in both indices.
+    */
+  sealed trait CETRange {
```

Comment:
I'm not sure this would add any value as all ranges are inspected symbolically by looking only at endpoints, whereas extending `NumericRange[Long]` would only introduce `Seq` and `Iterable` functionality which I don't currently see any use for